CHANGELOG.md

```diff
@@ -42,6 +54,195 @@ class Pet(Enum):
+ * Fix [override] error with no line number when argument node has no line number (Brian Schubert, PR [18122](https://github.com/python/mypy/pull/18122))
+ * Fix typos in `generics.rst` (yihong, PR [18110](https://github.com/python/mypy/pull/18110))
+ * Fix couple crashes in dmypy (Ivan Levkivskyi, PR [18098](https://github.com/python/mypy/pull/18098))
+ * Fix subtyping between Instance and Overloaded (Shantanu, PR [18102](https://github.com/python/mypy/pull/18102))
+ * Clean up new_semantic_analyzer config (Shantanu, PR [18071](https://github.com/python/mypy/pull/18071))
+ * Issue warning for enum with no members in stub (Shantanu, PR [18068](https://github.com/python/mypy/pull/18068))
+ * Fix enum attributes are not members (Terence Honles, PR [17207](https://github.com/python/mypy/pull/17207))
+ * Fix crash when checking slice expression with step 0 in tuple index (Brian Schubert, PR [18063](https://github.com/python/mypy/pull/18063))
+ * Allow union-with-callable attributes to be overridden by methods (Brian Schubert, PR [18018](https://github.com/python/mypy/pull/18018))
+ * Emit `[mutable-override]` for covariant override of attribute with method (Brian Schubert, PR [18058](https://github.com/python/mypy/pull/18058))
+ * Support ParamSpec mapping with functools.partial (Stanislav Terliakov, PR [17355](https://github.com/python/mypy/pull/17355))
+ * Fix approved stub ignore, remove normpath (Shantanu, PR [18045](https://github.com/python/mypy/pull/18045))
+ * Make `disallow-any-unimported` flag invertible (Séamus Ó Ceanainn, PR [18030](https://github.com/python/mypy/pull/18030))
+ * Filter to possible package paths before trying to resolve a module (falsedrow, PR [18038](https://github.com/python/mypy/pull/18038))
+ * Refactor type narrowing further (Jukka Lehtosalo, PR [18043](https://github.com/python/mypy/pull/18043))
+ * Refactor "==" and "is" type narrowing logic (Jukka Lehtosalo, PR [18042](https://github.com/python/mypy/pull/18042))
+ * Fix overlap check for ParamSpec types (Jukka Lehtosalo, PR [18040](https://github.com/python/mypy/pull/18040))
+ * Do not prioritize ParamSpec signatures during overload resolution (Stanislav Terliakov, PR [18033](https://github.com/python/mypy/pull/18033))
+ * Fix ternary union for literals (Ivan Levkivskyi, PR [18023](https://github.com/python/mypy/pull/18023))
+ * Improve for loop index variable type narrowing (Marc Mueller, PR [18014](https://github.com/python/mypy/pull/18014))
+ * Fix compatibility checks for conditional function definitions using decorators (Brian Schubert, PR [18020](https://github.com/python/mypy/pull/18020))
+ * Add timeout-minutes to ci config (Marc Mueller, PR [18003](https://github.com/python/mypy/pull/18003))
+ * Add initial support for new style TypeVar defaults (PEP 696) (Marc Mueller, PR [17985](https://github.com/python/mypy/pull/17985))
```

Comment:
Not sure if it's worth mentioning just yet, but the most common cases for TypeVar defaults should work with the new syntax now.

Reply:
I think it might be worth mentioning, just might need some help with the text. Something like this ok?

### Added initial support for new style TypeVar defaults (PEP 696)

Mypy now supports `TypeVar` defaults using the new syntax described in PEP 696, that was introduced in Python 3.13.

This is in its initial development and doesn't fully work yet for `ParamSpec`, `TypeVarTuple` and recursive `TypeVar` defaults.

This example from PEP 696 demonstrates the new syntax:
```python
T = TypeVar("T", default=int)  # This means that if no type is specified T = int

@dataclass
class Box(Generic[T]):
    value: T | None = None

reveal_type(Box())                      # type is Box[int]
reveal_type(Box(value="Hello World!"))  # type is Box[str]
```

Contributed by Marc Mueller (PR [17985](https://github.com/python/mypy/pull/17985))

Reply:
> ```python
> T = TypeVar("T", default=int)  # This means that if no type is specified T = int
> ```

This is the legacy syntax which mypy supports since `1.10`.
https://github.com/python/mypy/blob/v1.13.0/CHANGELOG.md#support-typealiastype-pep-695

With Python 3.13, this is enough now:
```py
@dataclass
class Box[T = int]:
    ...
```

--
Since the limitations apply equally to both syntax variants, it might be enough to highlight that the new syntax is supported now. The old changelog entry omitted the limitations as well. Something like this maybe?

--

### Added support for new style TypeVar Defaults (PEP 696)

Mypy now supports TypeVar defaults using the new syntax described in PEP 696, that was introduced in Python 3.13.

```py
@dataclass
class Box[T = int]:
    value: T | None = None

reveal_type(Box())                      # type is Box[int], since it's the default
reveal_type(Box(value="Hello World!"))  # type is Box[str]
```

Contributed by Marc Mueller (PR [17985](https://github.com/python/mypy/pull/17985))

Reply:
Perfect, thanks! (As you can tell I wasn't following too closely the changes in this area!)

Reply:
Updated!